### PR TITLE
Migrate API to new Cursor version

### PR DESCRIPTION
**Pull request by ktiays**

### CODE REVIEW

File: src/extension/index.ts [hunk 1]
```diff
@@ -100,6 +104,10 @@ export function activate(context: vscode.ExtensionContext) {
             new ChatPanelProvider(context)
         )
     );
+
+    // TODO: No need to refresh the token every time.
+    // If the token is still valid, there is no need to refresh it.
+    refreshToken();
```
Comment (unixzii) [inline src/extension/index.ts:110]: We need to pay attention to the case where tokens expire while the user is using the app.

File: crates/cursor-core/src/services/flagged_chunk.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
+use anyhow::{anyhow, Result};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone)]
+pub struct FlaggedChunk {
+    pub data: Vec<u8>,
+    pub flag: u8,
+}
+
+impl FlaggedChunk {
```
Comment (unixzii) [inline crates/cursor-core/src/services/flagged_chunk.rs:10]: Maybe we should call it `EnvelopedMessage` or something? I found it from here:  https://connect.build/docs/protocol.

File: crates/cursor-core/src/auth/mod.rs [hunk 3]
```diff
@@ -130,47 +133,46 @@ async fn polling(
...
-        match select(defer_abort_future, interval.next()).await {
-            Either::Left(_) => {
-                return Ok(None);
-            }
-            _ => {}
+        if let Either::Left(_) = select(defer_abort_future, interval.next()).await {
+            return Ok(None);
         }
-        let mut response = make_request_with_legacy(
+        let Ok(mut response)= make_request(
```
Comment (unixzii) [inline crates/cursor-core/src/auth/mod.rs:142]: Oops, fmt is not working 🫠

**Review by unixzii — CHANGES_REQUESTED** [review]

File: src/extension/index.ts (hunk 1 above)
Comment (ktiays) [reply src/extension/index.ts:110]: I think we can do it in the next PR as an improvement.

**Review by unixzii — APPROVED** [review]

**Pull request merged**